RegShoFINRADailyShorts.cs

```diff
@@ -0,0 +1,188 @@
+        {
+            while (ResultsNot404 && StartDate >= EndDate)
+            {
+                url = website + filenamePart1 + StartDate.ToString("yyyyMMdd") + filenamePart3;
+                destination = localDirectoryRoot + filenamePart1 + StartDate.ToString("yyyyMMdd") + filenamePart3;
+
+                if ((int) StartDate.DayOfWeek > 0 && (int) StartDate.DayOfWeek < 6) // Sunday & Saturday respectively
+                {
+                    Console.WriteLine("Pulling down file: {0}", url);
+
+                    using (WebClient client = new WebClient())
+                    {
+                        try
+                        {
+                            client.DownloadFile(url, destination);
+                            Console.WriteLine("\tSuccess! Downloading file {0}", url);
+                            ResultsFailCount = 0;
+                        }
+                        catch (WebException wex)
+                        {
+                            if (((HttpWebResponse) wex.Response).StatusCode == HttpStatusCode.NotFound)
+                            {
+                                Console.WriteLine("Unable to download {0}.  404 Not found.", url);
+
+                                ResultsFailCount++;
+                                if (ResultsFailCount >= 10)
+                                {
+                                    ResultsNot404 = false;
+                                }
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    Console.WriteLine("Date {0} falls on weekend.", StartDate.ToString("yyyyMMdd"));
+                }
+                StartDate = StartDate.AddDays(-1);
+
+                //Do not abuse those that give us the snackies.
+                Thread.Sleep(50);
+            }
+
+            Console.WriteLine("Loop ended. {0} failures occurred ending on {1}", ResultsFailCount, StartDate);
+        }
+
+
+
+        public void IngestDirectoryIntoMSSQL()
+        {
+            Console.WriteLine("Ingesting FINRA Shorts from directory: {0}", localDirectoryRoot);
+            string[] files = Directory.GetFiles(localDirectoryRoot, "*.txt", SearchOption.TopDirectoryOnly);
+
+            for (int i = 0; i < files.Length - 1; i++)
+            {
+                Console.WriteLine("Ingesting file: {0}", files[i]);
+                IngestSingleFileIntoMSSQL(files[i]);
+            }
+
+            Console.WriteLine("Operation completed.  Press any key to continue.");
+            Console.ReadLine();
+        }
+
+
+
+
+        public void IngestSingleFileIntoMSSQL(string f)
+        {
+            try
+            {
+                //Create Connection to SQL Server
+                SqlConnection SQLConnection = new SqlConnection();
+                SQLConnection.ConnectionString = @"Data Source = DESKTOP-1786GJA\SQLEXPRESS; Database=stunks; Integrated Security=true;";
```

Comment:
This should probably be configurable.

Reply:
Was just a test for access.  Didn't secret the information that would be sensitive in a real environment.  We can actually delete this.